components/automate-ui/src/app/pages/infra-proxy/chef-servers-details/chef-servers-details.component.ts

```diff
@@ -182,4 +188,33 @@ export class ChefServersDetailsComponent implements OnInit, OnDestroy {
   public closeDeleteModal(): void {
     this.deleteModalVisible = false;
   }
+
+  saveServer(): void {
+    this.saveSuccessful = false;
+    this.saving = true;
+    const updatedServer = {
+      id: this.server.id,
+      name: this.updateServerForm.controls.name.value.trim(),
+      description: this.updateServerForm.controls.description.value.trim(),
+      fqdn: this.updateServerForm.controls.fqdn.value.trim(),
+      ip_address: this.updateServerForm.controls.ip_address.value.trim()
+    };
+    this.store.dispatch(new UpdateServer({server: updatedServer}));
+    const pendingSave = new Subject<boolean>();
```

Comment:
We have recently identified this `pendingSave` (or `pendingAdd`, etc.) as an anti-pattern. You can find several PRs from about a month ago that have replaced this with a much cleaner, simpler approach.
Example PRs:
https://github.com/chef/automate/pull/2188
https://github.com/chef/automate/pull/2205
https://github.com/chef/automate/pull/2206

Please update accordingly.